### PR TITLE
Address Leaks and Go Vet Tool Issues

### PR DESCRIPTION
**Pull request by Mahmod**
This commit fixes various leaks and other errors identified by the go vet tool across the Golang codebase.

Please review this pull request carefully , since I only validated the suggested fixes through `go vet ./...` command .
I'm not familiar with the functional testing that assure the functionality of my suggestions .

Bug: b/309808700

### CODE REVIEW

File: pkg/client/httputils.go [hunk 1]
```diff
@@ -232,11 +235,16 @@ func (u *FilesUploader) Upload(files []string) error {
...
 			}
 		}
 	}
+
+	// Ensure all results are received to prevent goroutine leaks.
+	defer func() {
+		for range resultsChan {
+			// Drain the channel.
+		}
+	}()
```
Comment (jemoreira) [inline pkg/client/httputils.go:247]: This isn't needed. The loop above doesn't break until the channel is drained.

File: pkg/client/httputils.go [hunk 2]
```diff
@@ -232,11 +235,16 @@ func (u *FilesUploader) Upload(files []string) error {
 			if returnErr == nil {
 				returnErr = err
```
Comment (jemoreira) [inline pkg/client/httputils.go:236]: you can improve this line by making `returnErr` a `multierror` and appending each received error to it.

File: pkg/client/httputils.go [hunk 3]
```diff
@@ -218,6 +218,9 @@ func (u *FilesUploader) Upload(files []string) error {
 		return err
 	}
 	ctx, cancel := context.WithCancel(context.Background())
+	// ensure cancel is called even in the case of an error
```
Comment (jemoreira) [inline pkg/client/httputils.go:221]: Actually, the existing code only calls `cancel` when there are errors. This ensures it's called when there are NO errors too.

**Review by jemoreira — CHANGES_REQUESTED** [review]

**Comment by Mahmod** [comment]
This issue is resolved by @jemoreira in different Pr , no longer needed

**Pull request closed without merging**